Allow setting a URL to configure frontend and admin URLs (#13681)
keycloak/keycloak: docs/guides/src/main/server/hostname.adoc

Most of the time, it should be enough to set the `hostname-admin` option in order to change only the *host* of the administration console URLs:

<@kc.start parameters="--hostname-admin=<host>"/>

However, if you want to set not only the host but also a scheme, port, and path, you can set the `hostname-admin-url` option:

<@kc.start parameters="--hostname-admin-url=<scheme>://<host>:<port>/<path>"/>

Note that the `hostname-admin` and `hostname-admin-url` are mutually exclusive.

To reduce attack surface, the administration endpoints for Keycloak and the Admin Console should not be publicly accessible.
Therefore, you can secure them by using a reverse proxy.
For more information about which paths to expose using a reverse proxy, see the <@links.server id="reverseproxy"/> Guide.

== Example Scenarios
The following are more example scenarios and the corresponding commands for setting up a hostname.

Note that the `start` command requires setting up TLS. The corresponding options are not shown for example purposes. For more details,
see <@links.server id="enabletls"/> guide.

> === Example 1: Exposing the server behind a TLS termination proxy

Reviewer: Keycloak behind a reverse proxy

> === Example 1: Exposing the server behind a TLS termination proxy

Reviewer: ```suggestion
=== Exposing the server behind a TLS termination proxy
```

Reviewer: There's some mistakes in the numbering below, but I don't think it's needed to prefix these titles with "Example #:"

> In this example, the server is running behind a TLS termination proxy and publicly available from `https://mykeycloak`.

Reviewer: ```suggestion
In this example, the server is running behind a proxy that terminates TLS at the edge and publicly available from `https://mykeycloak`.
```

.Configuration:
<@kc.start parameters="--hostname=mykeycloak --proxy-edge"/>

> === Example 2: Exposing the server without a TLS termination proxy

Reviewer: ```suggestion
=== Example 2: Exposing the server without a proxy
```

In this example, the server is running without a proxy and exposed using a URL using HTTPS.

.Keycloak configuration:
<@kc.start parameters="--hostname-url=https://mykeycloak"/>

It is highly recommended using a TLS termination proxy in front of the server for security and availability reasons. For more details,
see the <@links.server id="reverseproxy"/> guide.

> === Example 3: Forcing backend endpoints to use the same URL the server is exposed

Reviewer: ```suggestion
=== Forcing backend endpoints to use the same URL the server is exposed
```

In this example, backend endpoints are exposed using the same URL used by the server so that clients always fetch the same URL
regardless of the origin of the request.

.Keycloak configuration:
<@kc.start parameters="--hostname=mykeycloak --hostname-strict-backchannel=true"/>

> === Example 3: Exposing the server using a port other than the default ports

Reviewer: ```suggestion
=== Exposing the server using a port other than the default ports
```
